sphinx/writers/text.py

```diff
@@ -29,6 +31,235 @@
 logger = logging.getLogger(__name__)
 
 
+class Cell:
+    """Represents a cell in a table.
+    It can span on multiple columns or on multiple lines.
+    """
+    def __init__(self, text="", rowspan=1, colspan=1):
+        self.text = text
+        self.wrapped = []
+        self.rowspan = rowspan
+        self.colspan = colspan
+        self.col = None
+        self.row = None
+
+    def __repr__(self):
+        return "<Cell {!r} {}v{}/{}>{}>".format(
+            self.text, self.row, self.rowspan, self.col, self.colspan
+        )
+
+    def __hash__(self):
+        return hash((self.col, self.row))
+
+    def wrap(self, width):
+        self.wrapped = my_wrap(self.text, width)
+
+
+class Table:
+    """Represents a table, handling cells that can span on multiple lines
+    or rows, like:
+    +-----------+-----+
+    | AAA       | BBB |
+    +-----+-----+     |
+    |     | XXX |     |
+    |     +-----+-----+
+    | DDD | CCC       |
+    +-----+-----------+
+
+    This class can be used in two ways:
+
+    - Either with absolute positions: call table[line, col] = Cell(...),
+      this overwrite an existing cell if any.
+
+    - Either with relative positions: call the add_row() and
+      add_cell(Cell(...)) as needed.
+
+    Cell spanning on multiple rows or multiple columns (having a
+    colspan or rowspan greater than one) are automatically referenced
+    by all the table cells they covers. This is a usefull
+    representation as we can simply check `if self[x, y] is self[x,
+    y+1]` to recognize a rowspan.
+
+    Colwidth is not automatically computed, it has to be given, either
+    at construction time, either during the table construction.
+
+    Example usage:
```

Comment:
not with `.. doctest::` ou `::` ?

Reply:
Oh, I'm surprised they use rst in their docstring. Why not.